app/models/detailed_guide.rb

```diff
@@ -151,12 +144,63 @@ def related_mainstream_content_valid?
     end
   end
 
+
   def additional_related_mainstream_content_valid?
     if additional_related_mainstream_content_url.present? && additional_related_mainstream_content_title.blank?
       errors.add(:additional_related_mainstream_content_title, "cannot be blank if an additional related URL is given")
     end
   end
 
+  def related_mainstream_found
+    return unless related_mainstream_requested?
+    fetch_related_mainstream_content_ids
+    add_errors_for_missing_related_mainstream
+    add_errors_for_missing_additional_related_mainstream
+  end
+
+  def add_errors_for_missing_related_mainstream
+    if @content_ids.count >= 1
+      errors.add(:related_mainstream_content_url, "This mainstream content could not be found") if @content_ids[0].nil?
+    end
+  end
+
+  def add_errors_for_missing_additional_related_mainstream
+    if @content_ids.count > 1
+      errors.add(:additional_related_mainstream_content_url, "This mainstream content could not be found") if @content_ids[1].nil?
+    end
+  end
+
+  def fetch_related_mainstream_content_ids
+    base_paths = [related_mainstream_base_path, additional_related_mainstream_base_path].compact
+    if need_to_fetch_related_mainstream?(base_paths)
+      @content_ids ||= []
+      lookup_content_ids(base_paths)
+    else
+      @content_ids ||= []
+    end
+  end
+
+  def need_to_fetch_related_mainstream?(base_paths)
+    base_paths.any? && @content_ids.nil?
+  end
+
+  def lookup_content_ids(base_paths)
+    base_paths.each do |base_path|
+      content_id = Whitehall.publishing_api_v2_client.lookup_content_id(base_path: base_path)
+      @content_ids << content_id
+    end
+  end
+
+  def related_mainstream_requested?
+    related_mainstream_content_url.present? || additional_related_mainstream_content_url.present?
+  end
+
+  def persist_content_ids
+    return if @content_ids.nil? || @content_ids.compact.empty?
+    RelatedMainstream.find_or_create_by!(edition_id: self.id, content_id: @content_ids[0])
```

Comment:
Only if they're creating a new edition. If they create a draft linking to /foo, save it, then change their mind and link to /bar, the link to /foo will still exist